src/services/submissionService.ts

```diff
@@ -56,8 +56,118 @@ function validateIdInput(id: string): Result<gqlTypes.Submission> {
     return validationResults
 }
 
+function validateSubmissionSearchFilters(filters: dbSchema.SubmissionSearchFilters): Result<gqlTypes.Submission[]> {
+    const validationResults: Result<gqlTypes.Submission[]> = {
+        hasErrors: false,
+        errors: []
+    }
+
+    const { orderBy, limit } = filters
+
+    if (orderBy?.some(order => order.fieldToOrder === 'id')) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.INVALID_SEARCH_OPTION,
+            httpStatus: 400
+        })
+    }
+
+    const validFields = [
+        'googleMapsUrl',
+        'healthcareProfessionalName',
+        'isUnderReview', 
+        'isApproved',
+        'isRejected', 
+        'createdDate', 
+        'updatedDate'
+    ]
+    
+    const invalidOrder = orderBy?.find(order => !validFields.includes(order.fieldToOrder))
+
+    if (invalidOrder) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.INVALID_ORDERBY_FIELD,
+            httpStatus: 400
+        })
+    }
+
+    if (orderBy && orderBy.length > 2) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'orderBy',
+            errorCode: ErrorCode.MAX_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (limit && (limit < 0)) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'limit',
+            errorCode: ErrorCode.MIN_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (limit && (limit > 1000)) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'limit',
+            errorCode: ErrorCode.MAX_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.googleMapsUrl === '' || (filters.googleMapsUrl && !filters.googleMapsUrl.trim())) {
```

Comment:
```suggestion
    if (!filters.googleMapsUrl || !filters.googleMapsUrl.trim()) {
```

I think this is equivalent 🤔

Reply:
@ermish I tried this and found it throws an error if `googleMapsUrl` is falsy, meaning it’ll thrown an error when it’s not provided as a search filter. The original throws an error only if `googleMapsUrl` is chosen as a search filter but contains an empty string or whitespace that becomes empty after trimming. How do we want to handle this?